config/initializers/csp.rb

```diff
@@ -1 +1,76 @@
-GovukContentSecurityPolicy.configure
+# GovukContentSecurityPolicy.configure
+# THIS A TEMPORARY COPY OF THE POLICY FROM THE GEM ABOVE
+# FOR THE TEMPORARY TEST WE NEED TO ADD A NEW DOMAIN FOR IMAGES (LINE 30)
+# WILL BE REVERTED BACK TO THE GEM ONCE THE TESTS FINISHES
+
+GOVUK_DOMAINS = [
+    '*.publishing.service.gov.uk',
+    "*.#{ENV['GOVUK_APP_DOMAIN_EXTERNAL'] || ENV['GOVUK_APP_DOMAIN'] || 'dev.gov.uk'}",
+    "*.dev.gov.uk"
+  ].uniq.freeze
+
+GOOGLE_ANALYTICS_DOMAINS = %w(www.google-analytics.com
+                              ssl.google-analytics.com
+                              stats.g.doubleclick.net).freeze
+
+Rails.application.config.content_security_policy_report_only = ENV.include?("GOVUK_CSP_REPORT_ONLY")
+
+Rails.application.config.content_security_policy do |policy|
+  # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/default-src
+  policy.default_src :https, :self, *GOVUK_DOMAINS
+
+  # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src
+  policy.img_src :self,
+                :data, # Base64 encoded images
+                *GOVUK_DOMAINS,
+                *GOOGLE_ANALYTICS_DOMAINS, # Tracking pixels
+                # Some content still links to an old domain we used to use
+                "assets.digital.cabinet-office.gov.uk",
+                # For a Verify enhanced hint live test (temporary)
+                "gds-verify-frontend-assets.s3.amazonaws.com"
```

Comment:
This feels very hacky. In an ideal world, I'd like for that extra domain to be configured in `govuk_app_config`. Would that be the plan for if the tests succeed goes live permanently, or getting them onto GOV.UK's asset mangement system somehow?

Reply:
yes, it does (and it's not the only hacky thing :( ).. the `ENV['GOVUK_APP_DOMAIN_EXTERNAL']` could be used to pass it in, but that will also allow scripts and everything.. we only need images... If this was going to live, the CSP would have to be considered. At its current form it will get teared down...

Reply:
By "this" I meant "reproducing the whole of CSP for this one thing that deviates from the gem for everything in this app now". :-)